src/components/Dialogs/PreferenceDialog/PreferenceDialogComponent.tsx

```diff
@@ -18,8 +36,37 @@ export class PreferenceDialogComponent extends React.Component<{ appStore: AppSt
             <div className="panel-container">
                 <FormGroup inline={true} label="Scaling">
                     <ScalingComponent
-                        selectedItem={preferenceStore.getScaling()}
-                        onItemSelect={preferenceStore.setScaling}
+                        selectedItem={this.scaling}
+                        onItemSelect={(selected) => { preferenceStore.setScaling(selected); this.scaling = selected; }}
+                    />
+                </FormGroup>
+                <FormGroup inline={true} label="Color map">
+                    <ColormapComponent
+                        selectedItem={this.colormap}
+                        onItemSelect={(selected) => { preferenceStore.setColormap(selected); this.colormap = selected; }}
+                    />
+                </FormGroup>
+                <FormGroup inline={true} label="Percentile ranks">
+                    <PercentilSelect
+                        activeItem={this.percentile}
+                        onItemSelect={(selected) => { preferenceStore.setPercentile(selected); this.percentile = selected; }}
```

Comment:
is this done to avoid re-reading the value from local storage on each re-render?

Reply:
if I replace this.percentile with this.props.appStore.preferenceStore.getPercentile().toString() to reduce that observable variable, the content of the selected item will not be updated to UI (it keeps the initially loaded value), so the solution I found is to make it observable and update it in onItemSelect/onChange, is there a better way for this?